ckan/controllers/package.py

```diff
@@ -224,15 +224,25 @@ def pager_url(q=None, page=None):
                        'user': c.user, 'for_view': True,
                        'auth_user_obj': c.userobj}
 
-            if package_type and package_type != 'dataset':
+            # Unless changed via config options, don't show other dataset
+            # types any search page. Potential alternatives are do show them
+            # on the default search page (dataset) or on one other search page
+            search_all_type = config.get('ckan.search.show_all_types', 'false')
+            search_all = False
+
+            try:
+                # If the "type" is set to True or False, convert to bool
+                search_all = asbool(search_all_type)
```

Comment:
won't this show all types on all types if show_all_types = true? We should maintain the current behaviour of only showing all on the 'dataset' type